File: app/src/main/java/com/triangleleft/flashcards/di/main/MainPageModule.java
package com.triangleleft.flashcards.di.main;

import com.triangleleft.flashcards.di.scope.ActivityScope;
import com.triangleleft.flashcards.ui.main.MainPresenter;
import com.triangleleft.flashcards.ui.vocabular.VocabularyNavigator;

import dagger.Module;
import dagger.Provides;

@Module
public class MainPageModule {

    @ActivityScope
    @Provides
    public VocabularyNavigator vocabularNavigator(MainPresenter presenter) {
        return presenter;
    }
}

File: app/src/main/java/com/triangleleft/flashcards/di/main/MainPageModule.kt
package com.triangleleft.flashcards.di.main

import com.triangleleft.flashcards.service.vocabular.VocabularyWord
import dagger.Module
import dagger.Provides
import io.reactivex.Observable
import io.reactivex.Observer
import io.reactivex.subjects.BehaviorSubject
import javax.inject.Named

@Module
class MainPageModule {

    private val wordSelections = BehaviorSubject.create<VocabularyWord>()

    @Provides
    @Named(WORD_SELECTIONS)
    fun wordObserver(): Observer<VocabularyWord> {
        return wordSelections
    }

    @Provides
    @Named(WORD_SELECTIONS)
    fun wordSelections(): Observable<VocabularyWord> {
        return wordSelections
    }

    companion object {
        const val WORD_SELECTIONS = "wordSelections"
    }

}
